### PR TITLE
rmapi: init at v0.0.7

### PR DESCRIPTION
**Pull request by Enteee**
###### Motivation for this change

Add the [rMAPI](https://github.com/juruen/rmapi) client for the reMarkable cloud.

Related pull request to [juruen/rmapi](https://github.com/juruen/rmapi/pull/78)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by Enteee** [comment]
I am struggling to understand the CI error.

**Comment by Enteee** [comment]
when I run :
```
GC_INITIAL_HEAP_SIZE=4g nix-env -f ./outpaths.nix -qaP --no-name --out-path --arg checkMeta true pkga.rmapi 2>&1 | tee out-paths
```
I get:
```
error: cannot import '/nix/store/j6xvkiilmdb23hjxgp3j9pri1qz2w9sc-source/derivation.nix', since path '/nix/store/y74warfhqa098y60zv40v312lf9kfnz5-source.drv' is not valid, at /home/user/workspace/nixpkgs/lib/customisation.nix:119:45
(use '--show-trace' to show detailed location information)
```

File: pkgs/applications/office/rmapi/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ callPackage, fetchFromGitHub, buildGoModule, buildEnv, stdenv}:
+
+let
+  pname = "rmapi";
+  version = "0.0.7";
+
+  repo = fetchFromGitHub {
+    owner = "juruen";
+    rev = "tags/v${version}";
```
Comment (doronbehar) [inline pkgs/applications/office/rmapi/default.nix]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/office/rmapi/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (doronbehar) [inline pkgs/applications/office/rmapi/default.nix]:
```suggestion
```

File: pkgs/applications/office/rmapi/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+# The idea of the following was to keep the derivation.nix and deps.nix
+# in juren/rmapi. Sadly, this does not work becuase hydra does not support
+# import-from-derivation. For this reason, we have to manually copy the
+# two files into nixpkgs.
+# Reference:
+# * https://discourse.nixos.org/t/is-importing-a-derivation-from-another-repository-bad-practice/5016/4?u=enteee
+#  callPackage "${repo}/derivation.nix" {
+
+  callPackage ./derivation.nix {
+    buildGoModule = super: buildGoModule (super // {
```
Comment (doronbehar) [inline pkgs/applications/office/rmapi/default.nix]: I'm not sure I understand what's "import from derivation", but I think the meaning is that using simply `buildGoModule` as is doesn't work?

Comment (Enteee) [reply pkgs/applications/office/rmapi/default.nix]: No that's not what IFS means. I updated the pull request with a link to the wiki which explains IFS derivations. I am nor 100% sure if I should leave this comment or if I should get rid of it entirely.

**Comment by NickHu** [comment]
Possibly this pull request is superceded by mine: https://github.com/NixOS/nixpkgs/pull/85898

That'll teach me to make a PR without searching for it first (again)...

**Comment by Enteee** [comment]
@NickHu: I have already mentioned this pr on the discourse thread, so I am not quite sure what to do about this so that it can get merged. But I would suggest the following:

# I will bump the version of this to match the current rmapi version
# then you can possibly close your pr?..
# you review this pr and mention it again on the discourse thread

what do you think about this?

**Comment by Enteee** [comment]
replaced by:  #85898

**Pull request closed without merging**